semgrep-core/src/osemgrep/cli_scan/Semgrep_dashdash_config.ml

```diff
@@ -10,20 +14,82 @@
    a possible .semgrep).
 
    Partially translated from config_resolver.py
+
+   TODO:
+    - handle the registry-aware jsonnet format (LONG)
+    - lots of stuff ...
 *)
 
 (*****************************************************************************)
 (* Types *)
 (*****************************************************************************)
 
+type config_kind =
+  | File of Common.filename
+  | Dir of string
+  | R of registry_kind
+
+and registry_kind =
+  (* r/... *)
+  | Registry of string
+  (* p/... *)
+  | Pack of string
+  (* s/... *)
+  | Snippet of string
+  (* pad:basic *)
+  | SavedSnippet of string (* username *) * string (* snippetname *)
+[@@deriving show]
+
 (*****************************************************************************)
 (* Helpers *)
 (*****************************************************************************)
+let config_kind_of_config_str config_str =
+  match config_str with
+  | s when s =~ "^r/\\(.*\\)" -> R (Registry (Common.matched1 s))
+  | s when s =~ "^p/\\(.*\\)" -> R (Pack (Common.matched1 s))
+  | s when s =~ "^s/\\(.*\\)" -> R (Snippet (Common.matched1 s))
+  | s when s =~ "^\\(.*\\):\\(.*\\)" ->
+      let user, snippet = Common.matched2 s in
+      R (SavedSnippet (user, snippet))
+  (* TODO  | file when Sys.is_dir file  -> Dir file *)
+  | file when Sys.file_exists file -> File file
+  | _else_ -> failwith (spf "not a valid --config string: %s" config_str)
+
+let url_of_registry_kind rkind =
+  (* TODO: go through curl interface for now (c/) *)
+  let prefix = "https://semgrep.dev/c" in
+  match rkind with
+  | Registry s -> spf "%s/r/%s" prefix s
+  | Pack s -> spf "%s/r/%s" prefix s
+  | Snippet s -> spf "%s/s/%s" prefix s
+  | SavedSnippet (user, snippet) -> spf "%s/%s:%s" prefix user snippet
+
+let load_rules_from_file file =
+  logger#debug "loading local config from %s" file;
+  Parse_rule.parse_and_filter_invalid_rules file
 
 (*****************************************************************************)
 (* Entry point *)
 (*****************************************************************************)
 
-let rules_from_dashdash_config config =
+open Lwt.Syntax
+open Cohttp_lwt_unix
```

Comment:
I would use a module alias instead of an `open` here. Not being familiar with Cohttp I would prefer to easily identify what types and functions belong to it. For example, I skimmed the code and I was wondering what that `Client` module was.

Reply:
good idea

Reply:
for Lwt.syntax we have to open it, to benefit from the let*, but I can add a comment about it.